scenario/state.py

```diff
@@ -140,6 +140,69 @@ def copy(self) -> "Self":
         return copy.deepcopy(self)
 
 
+class CloudAuthType(str, Enum):
+    access_key_auth_type = "access-key"
+    instance_role_auth_type = "instance-role"
+    user_pass_auth_type = "userpass"
+    o_auth1_auth_type = "oauth1"
+    o_auth2_auth_type = "oauth2"
```

Comment:
Somehow `OAUTH` seems more natural than `O_AUTH` (the spec is called OAuth):
```suggestion
    OAUTH1 = "oauth1"
    OAUTH2 = "oauth2"
```